libsolidity/ast/Types.cpp

```diff
@@ -561,6 +804,55 @@ shared_ptr<IntegerType const> IntegerConstantType::integerType() const
 		);
 }
 
+shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
+{
+	bool negative = (m_value < 0);
+	unsigned fractionalBits = 0;
+	unsigned integerBits = 0;
+	rational value = m_value;
+	bigint transitionValue = bigint(1) << 256;
+	rational maxValue = rational(transitionValue);
+
+	if (!negative)
+	{
+		maxValue -= 1;
+		integerBits = bytesRequired(wholeNumbers()) * 8;
+	}
+	else
+	{
+		value = abs(value);
+		if (wholeNumbers() > 0)
+			transitionValue = ((0 - wholeNumbers()) - 1) << 1;
+		else
+			transitionValue = 0;
+		integerBits = bytesRequired(transitionValue) * 8;
+	}
+
+	while (value * 0x100 <= maxValue && value.denominator() != 1 && fractionalBits < 256 - integerBits)
```

Comment:
If there is an error in the implementation here, it might very well be that we will never notice.
Please compare "there might be an error we never notice" to "looping 20 or 32 times".